src/charm.py

```diff
@@ -1390,11 +1394,6 @@ def _reconciliation(self, _event: EventBase) -> None:
             _event: Event fired by juju on WordPress charm related state change.
         """
         logger.info("Start reconciliation process, triggered by %s", _event)
-        if not self.state.started:
```

Comment:
You can probably remove the logic in line 150
```
self.state.set_default(
    started=False,
)
```
and in line 186
```
def _on_start(self, _event: StartEvent):
    """Record if the start event is emitted."""
    self.state.started = True
```

Reply:
Yes, that was originally used to prevent a problem in Juju, causing applications not to complete the `start` event within the grace period to be rescheduling by Kubernetes. However, I just found out that the reconciliation process is no longer dependent on the `start` event, and I don't know when that was removed. But I think it's okay for now, and @yanksyoon will have some pull requests to remove those entirely, along with the database connectivity test and deferring for storage ready.